neo/rawio/spikeglxrawio.py

```diff
@@ -292,33 +293,113 @@ def _event_count(self, event_channel_idx, block_index=None, seg_index=None):
         return timestamps.size
 
     def _get_event_timestamps(self, block_index, seg_index, event_channel_index, t_start=None, t_stop=None):
+        #TODO: fix seg_index usage, currently hardcoded for first segment
         timestamps, durations, labels = [], None, []
         info = self.signals_info_dict[0, "nidq"]  # There are no events that are not in the nidq stream
         dig_ch = info["digital_channels"]
         if len(dig_ch) > 0:
             event_data = self._events_memmap
-            channel = dig_ch[event_channel_index]
+            channel = dig_ch[event_channel_index] # 'XD0', 'XD1', etc.
             ch_idx = 7 - int(channel[2:])  # They are in the reverse order
```

Comment:
This calculation of channel index is also something that I had hard-coded and will work only when there are exactly 8 digital channels saved (which was the cases for our setup, but isn't generally true); but not when you save fewer channels or have I/O devices that have more than 8 digital channels.

Reply:
Thanks for pointing this out, this is something to fix.